nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java

```diff
@@ -105,6 +107,19 @@ public BulkStatus getStatus(String bulkId) {
         return status;
     }
 
+    @Override
+    public boolean await(String bulkId, long timeout, TimeUnit unit) throws InterruptedException {
+        long deadline = System.currentTimeMillis() + unit.toMillis(timeout);
+        KeyValueStore kvStore = getKvStore();
+        do {
+            if (COMPLETED.toString().equals(kvStore.getString(bulkId + STATE))) {
+                return true;
+            }
+            Thread.sleep(200);
```

Comment:
We said that the counter stat computation should update the state ~2x per second, so you increase to 500ms